### PR TITLE
[MO][OPS-11744] Integrate Notify case management API

### PR DESCRIPTION
**Pull request by Lyrain**

### CODE REVIEW

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 1]
```diff
@@ -181,9 +182,66 @@ class WeAreVerifyingYourBankAccountController @Inject() (
...
+
+    val request: CaseManagementRequest = {
+      val r: CaseManagementRequest = CaseManagementRequest(
+        clientUId             = clientUId,
+        clientSystemId        = ClientSystemId("MDTP"),
+        nino                  = journey.getNino,
+        bankSortCode          = sortCode,
+        bankAccountNumber     = accountNumber,
+        bankAccountName       = bankAccountName,
+        designatedAccountFlag = 1, // TODO: Confirm what this should be
```
Comment (Lyrain) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:208]: Not sure what this should be.

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 2]
```diff
@@ -181,9 +182,66 @@ class WeAreVerifyingYourBankAccountController @Inject() (
...
+        contact               = List(
+          CaseManagementContact(
+            `type`    = PersonType.Customer,
+            firstName = journey.getTraceIndividualResponse.firstForename.getOrElse(""),
+            surname   = journey.getTraceIndividualResponse.surname,
+            address   = List() // TODO: Confirm what this should be
```
Comment (Lyrain) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:214]: Are we pulling the address from Ecospend or NPS?

Comment (barnesjake) [reply app/controllers/WeAreVerifyingYourBankAccountController.scala:214]: Good question, not sure

**Review by barnesjake — APPROVED** [review]

File: app/controllers/WeAreVerifyingYourBankAccountController.scala (hunk 2 above)
Comment (barnesjake) [reply app/controllers/WeAreVerifyingYourBankAccountController.scala:214]: We should probably raise a ticket to make sure we do this (whichever one) before e2e

File: app/controllers/WeAreVerifyingYourBankAccountController.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/controllers/WeAreVerifyingYourBankAccountController.scala:208]: I think this should be `0`. `1` means that the account is designated. To be confirmed :crossed_fingers:

File: app/controllers/WeAreVerifyingYourBankAccountController.scala (hunk 2 above)
Comment (pawelpanasewicz) [reply app/controllers/WeAreVerifyingYourBankAccountController.scala:214]: This is just the first implementation of this API without prior full spike regards data mapping. Please leave TODOs whenever it's not clear what to pass in.

File: app/edh/EdhConnector.scala [hunk 3]
```diff
@@ -52,6 +52,27 @@ class EdhConnector @Inject() (
...
+
+    httpClient
+      .POST[CaseManagementRequest, Either[UpstreamErrorResponse, HttpResponse]](
+        url     = notifyCaseManagementUrl(clientUId),
+        body    = request,
+        headers = makeHeaders()
+      ).map {
+          case Right(_)    => ()
+          case Left(error) => throw error
+        }
```
Comment (pawelpanasewicz) [inline app/edh/EdhConnector.scala:73]: I'm suspicious about those reads. Will this always fail for non 2xx responses? For example if response http status code is 400, will it result in something like `Right(HttpResponse(400)) ` or throw exception?

File: app/edh/EdhConnector.scala [hunk 4]
```diff
@@ -52,6 +52,27 @@ class EdhConnector @Inject() (
 
   }
 
+  private def notifyCaseManagementUrl(clientUId: ClientUId): String = baseUrl +
+    s"/risking/exceptions/${clientUId.value}"
+
+  def notifyCaseManagement(
```
Comment (pawelpanasewicz) [inline app/edh/EdhConnector.scala:58]: I think this is not EDH but Case Management system. 
Could you move it to seperate class (`casemanagement.CaseManagementConnector`) and also corresponding model classes to dedicated package (`casemanagement.model.*`)

File: app/edh/EdhConnector.scala (hunk 3 above)
Comment (pawelpanasewicz) [reply app/edh/EdhConnector.scala:73]: Could you consider such reads:


```
    implicit val readUnit: HttpReads[Unit] = HttpReads.ask.map {
      case (method, url, response) => response.status match {
        case 200 => ()
        case other => throw UpstreamErrorResponse(
          message    = HttpErrorFunctions.upstreamResponseMessage(method, url, other, response.body),
          statusCode = response.status,
          reportAs   = if (HttpErrorFunctions.is4xx(response.status)) INTERNAL_SERVER_ERROR else BAD_GATEWAY,
          headers    = response.headers
        )
      }
    }
```

**Pull request merged**